libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -636,7 +636,8 @@ bool AP_Arming::gps_checks(bool report)
             }
         }
 
-        if (!AP::ahrs().home_is_set()) {
+        // Plane uses an option to allow users to fly without a home.
+        if (!AP::ahrs().allows_arming_without_home() && !AP::ahrs().home_is_set()) {
```

Comment:
Perhaps we should move this check to the vehicle. On plane, I would like should check whether our current mode requires home. If it does, and we don't have a home yet, arming should be blocked.

Reply:
![image](https://github.com/user-attachments/assets/309f06c5-f475-4ca1-8a37-ab2e268a8a18)

As implemented, you can switch to LOITER while disarmed, but won't be allowed to arm till you set home.

Reply:
>     * AHRS home is the RTL point, position estimate is from EKF origin. It should be safe to treat having a home and having an ekf origin as the same.

... or, you know, you could check to see if you have a navigation origin instead....

If your failsafes include "RTL", then you might go further and require home to be set then.

Reply:
Are you saying we might want to add two methods to the mode class to prevent mode entry:
1. The one I currently have of `requires_home`, which uses `AP_AHRS::home_is_set()`
1. Another of `requires_navigation_origin`, which uses `AP_AHRS::get_origin()`

I agree that some of the modes only need the navigation origin, and I'm happy to add that to that check to mode entry if we are ok with the flash.